expression/aggregation/base_func.go

```diff
@@ -184,6 +184,10 @@ func (a *baseFuncDesc) typeInfer4MaxMin(ctx sessionctx.Context) {
 		a.Args[0] = expression.BuildCastFunction(ctx, a.Args[0], tp)
 	}
 	a.RetTp = a.Args[0].GetType()
+	if (a.Name == ast.AggFuncMax || a.Name == ast.AggFuncMin) && !a.RetTp.Hybrid() {
```

Comment:
Why do we need to check `!a.RetTp.Hybrid()`?

Reply:
1. Hybird type is special
2. It seems like someone has modify `a.RetTp = a.Args[0].GetType()` when process `Bit` type.